decidim-core/lib/decidim/reportable.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+require "active_support/concern"
+
+module Decidim
+  # A concern with the features needed when you want a model to be reportable
+  module Reportable
+    extend ActiveSupport::Concern
+
+    included do
+      has_one :moderation, as: :reportable, foreign_key: "decidim_reportable_id", foreign_type: "decidim_reportable_type", class_name: "Decidim::Moderation"
+      has_many :reports, through: :moderation
+
+      scope :reported, -> { left_outer_joins(:moderation).where(Decidim::Moderation.arel_table[:report_count].gt 0) }
+      scope :hidden, -> { left_outer_joins(:moderation).where.not(Decidim::Moderation.arel_table[:hidden_at].eq nil) }
+      scope :not_hidden, -> { left_outer_joins(:moderation).where(Decidim::Moderation.arel_table[:hidden_at].eq nil) }
```

Comment:
We should add an index to `report_count` and `hidden_at` at each reportable table